.github/workflows/continuous-integration-workflow.yml

```diff
@@ -11,41 +11,21 @@ on:
 
 jobs:
   CI:
-    continue-on-error: ${{ matrix.config.continue-on-error == 'true' }}
     strategy:
-      matrix:
-        config:
-          - {dockerfile: 'fedora',   tag: 'latest'}
-          - {dockerfile: 'fedora',   tag: 'rawhide',          build_args: 'TAG=rawhide', continue-on-error: 'true'}
-          - {dockerfile: 'ubuntu',   tag: 'latest'}
-          - {dockerfile: 'ubuntu',   tag: 'rolling',          build_args: 'TAG=rolling'}
-          - {dockerfile: 'ubuntu',   tag: 'devel',            build_args: 'TAG=devel', continue-on-error: 'true'}
-          - {dockerfile: 'opensuse', tag: 'latest'}
-          - {dockerfile: 'fedora',   tag: 'intel',            build_args: 'TAG=37,INTEL=yes', continue-on-error: 'true'}
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout out code
-        uses: actions/checkout@v3
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
-      - name: Login to Github Container Registry
-        uses: docker/login-action@v2
-        if: github.repository_owner == 'kokkos' && ( github.event_name == 'push' ||  github.event_name == 'schedule' )
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }} 
-      - name: Convert build_args
-        id: build_args
-        run: |
-          echo "args<<EOF" >> $GITHUB_OUTPUT 
-          echo ${{ matrix.config.build_args }} | sed 's/,/\n/' >> $GITHUB_OUTPUT
-          echo "EOF" >> $GITHUB_OUTPUT
-      - name: Build and Push Docker images for Github Container Registry
-        uses: docker/build-push-action@v3        
-        with:
-          tags: ghcr.io/kokkos/ci-containers/${{ matrix.config.dockerfile }}:${{ matrix.config.tag }}
-          file: ${{ matrix.config.dockerfile }}
-          build-args: ${{ steps.build_args.outputs.args }}
-          pull: true
-          push: ${{ github.repository_owner == 'kokkos' && ( github.event_name == 'push' ||  github.event_name == 'schedule' ) }}
+        matrix:
+          config:
+            - {dockerfile: 'fedora',   tag: 'latest'}
+            - {dockerfile: 'fedora',   tag: 'rawhide',  build_args: 'TAG=rawhide',  continue-on-error: 'true'}
+            - {dockerfile: 'ubuntu',   tag: 'latest'}
+            - {dockerfile: 'ubuntu',   tag: 'rolling',  build_args: 'TAG=rolling'}
+            - {dockerfile: 'ubuntu',   tag: 'devel',    build_args: 'TAG=devel',    continue-on-error: 'true'}
+            - {dockerfile: 'opensuse', tag: 'latest'}
+            - {dockerfile: 'fedora',   tag: 'intel',    build_args: 'TAG=37,INTEL=yes',    cmake_args: '-DCMAKE_CXX_COMPILER=icpc',  continue-on-error: 'true'}
+            - {dockerfile: 'fedora',   tag: 'intel',    build_args: 'TAG=37,INTEL=yes',    cmake_args: '-DCMAKE_CXX_COMPILER=icpx -DCMAKE_CXX_FLAGS=-fp-model=precise',  continue-on-error: 'true'}
+    uses: ./.github/workflows/build-and-push.yml
```

Comment:
Is there a reason to hide all details in another yml file?

Reply:
This was supposed to improve clarity and clearly separate different parts of the workflow. Given that the end result is not very complicated, this separation doesn't seem like a must have.

Reply:
Yeah I would prefer having it all in one file again!

Reply:
Done.